packages/flutter_tools/lib/src/commands/build_linux.dart

```diff
@@ -39,12 +61,23 @@ class BuildLinuxCommand extends BuildSubCommand {
   Future<FlutterCommandResult> runCommand() async {
     final BuildInfo buildInfo = await getBuildInfo();
     final FlutterProject flutterProject = FlutterProject.current();
+    final TargetPlatform targetPlatform =
+        getTargetPlatformForName(stringArg('target-platform'));
+    final bool needCrossBuild =
+        getNameForHostPlatformArch(_operatingSystemUtils.hostPlatform)
+            != getNameForTargetPlatformArch(targetPlatform);
+
     if (!featureFlags.isLinuxEnabled) {
       throwToolExit('"build linux" is not currently supported.');
     }
     if (!globals.platform.isLinux) {
       throwToolExit('"build linux" only supported on Linux hosts.');
     }
+    // Cross-building for x64 targets on arm64 hosts is not supported.
```

Comment:
Could you add an additional check for x64 host arm64 crossbuild, since it's currently blocked by https://github.com/flutter/flutter/issues/74929. Something like:

```
// TODO(fujino): https://github.com/flutter/flutter/issues/74929
if (_operatingSystemUtils.hostPlatform == HostPlatform.linux_x64 && targetPlatform == TargetPlatform.linux_arm64) {
  throwToolExit('Cross-build from Linux x64 host to Linux arm64 target is not currently supported.');
}
```

Reply:
Actually, that change will make the rest of this function dead code. Nevermind, you can leave it.

Reply:
I've added that.